src/test/java/engineering/swat/watch/TortureTests.java

```diff
@@ -0,0 +1,387 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.LocalTime;
+import java.util.Random;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import engineering.swat.watch.WatchEvent.Kind;
+
+class TortureTests {
+
+    private final Logger logger = LogManager.getLogger();
+
+    private TestDirectory testDir;
+
+    @BeforeEach
+    void setup() throws IOException {
+        testDir = new TestDirectory();
+    }
+
+    @AfterEach
+    void cleanup() throws IOException {
+        if (testDir != null) {
+            testDir.close();
+        }
+    }
+
+    private final class IOGenerator {
+        private final Set<Path> pathsWritten = ConcurrentHashMap.<Path>newKeySet();
+        private final Semaphore startRunning = new Semaphore(0);
+        private final Semaphore stopRunning = new Semaphore(0);
+        private final Semaphore done = new Semaphore(0);
+        private final int jobs;
+
+        IOGenerator(int jobs, Path root, Executor exec) {
+            this.jobs = jobs;
+            for (int j = 0; j < jobs; j++) {
+                startJob(root.resolve("run" + j), new Random(j), exec);
+            }
+        }
+
+        private final static int BURST_SIZE = 1000;
+
+        private void startJob(final Path root, Random r, Executor exec) {
+            exec.execute(() -> {
+                try {
+                    startRunning.acquire();
+                    var end = LocalTime.now().plus(TestHelper.NORMAL_WAIT.multipliedBy(2));
+                    while (!stopRunning.tryAcquire(100, TimeUnit.MICROSECONDS)) {
+                        if (LocalTime.now().isAfter(end)) {
+                            break;
+                        }
+                        try {
+                            // burst a bunch of creates creates and then sleep a bit
+                            for (int i = 0; i< BURST_SIZE; i++) {
+                                var file = root.resolve("l1-" + r.nextInt(1000))
+                                    .resolve("l2-" + r.nextInt(100))
+                                    .resolve("l3-" + r.nextInt() + ".txt");
```

Comment:
Nit: Why not use `i` (uniqueness guaranteed) instead of a random number (clashes improbable but possible, making it flaky)?

Reply:
because I'm fine with it sometimes randomly colliding. actually, we want that.